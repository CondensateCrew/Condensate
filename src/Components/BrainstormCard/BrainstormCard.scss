@import "../../css/variables.scss";

.bs-card {
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $big-blocks-border-radius;
  break-inside: avoid;
  display: inline-block;
  padding: $big-block-padding-space;
  margin-bottom: $medium-space;
  img {
    cursor: pointer;
    &:hover {
      transform: scale(1.1);
    }
    &:active {
      transform: scale(0.8);
    }
  }
  &>header {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    position: relative;
    h4 {
      text-align: left;
      line-height: 1.4em;
      width: 90%;

    }
    &>img {
      width: $small-text;
      margin-left: $tiny-space;
      margin-top: $tiny-space;
    }
  }
  h5 {
    cursor: pointer;
    opacity: 0.5;
    padding: $extra-small-space 0 $medium-space;
    &:hover {
      opacity: 1;
    }
  }
  p {
    font-size: $tiny-text;
  }
  &>p {
    line-height: 1.5em;
    text-align: left;
  }
  &>footer {
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    align-items: flex-end;
    flex-wrap: wrap;
    margin-top: $medium-space;
    height: auto;
    p {
      background: $primary-gradient;
      box-shadow: $external-shadow;
      border-radius: $round-border-radius;
      padding: $button-padding-space;
      margin: 0 $tiny-space;
      &.genius-label {
        background: $genius-gradient;
      }
    }
  }
}
