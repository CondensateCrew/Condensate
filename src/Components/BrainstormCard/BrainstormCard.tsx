import './BrainstormCard.scss';
import React, { useState } from 'react';
import { Category, Action } from 'interfaces';
import menu from 'assets/menu-dot.svg';
import MenuBlock from 'Containers/MenuBlock/MenuBlock';

interface Props {
  id: number,
  question: string,
  response: string,
  action: Action,
  isGenius: boolean,
  categories: Category[]
}

const BrainstormCard: React.FC<Props> = (props) => {
  const [ isClicked, setIsClicked ] = useState<boolean>(false);
  const { id, question, response, action, isGenius, categories } = props;
  const catgs = categories.map((ctg: Category) => (
    <p key={`ctg-${ctg.name.toLowerCase()}`}>{ctg.name}</p>
  ));

  const showMenu = (): void => setIsClicked(true);

  return (
    <div className='bs-card'>
      <header>
        <h4>{question}</h4>
        {
          !isClicked
            ? <img src={menu} alt="menu" onClick={showMenu}/>
            : <MenuBlock {...{setIsClicked, id}}/>
        }
      </header>
      <h5>{action.action}</h5>
      <p>{response}</p>
      <footer>
        { isGenius && <p className="genius-label">Genius</p> }
        { catgs }
      </footer>
    </div>
  )
}

export default BrainstormCard;
