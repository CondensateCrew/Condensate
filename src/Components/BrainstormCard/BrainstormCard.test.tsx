import React from 'react';
import BrainstormCard from './BrainstormCard';
import { shallow } from 'enzyme';

describe('BrainstormCard component', () => {
  let wrapper: any;

  const mockProps = {
    id: 1,
    question: 'What?',
    response: 'Idea',
    action: {
      id: 1,
      action: 'Build an app'
    },
    isGenius: true,
    categories: [{
      id: 1,
      name: 'Tech'
    }]
  };

  beforeEach(() => {
    wrapper = shallow(<BrainstormCard {...mockProps}/>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if menu is clicked', () => {
    wrapper.find('img[alt="menu"]').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if brainstorm is not genius', () => {
    const wrapper = shallow(
      <BrainstormCard {...mockProps} isGenius={false}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if brainstorm is not genius and menu is clicked', () => {
    const wrapper = shallow(
      <BrainstormCard {...mockProps} isGenius={false}/>
    );
    wrapper.find('img[alt="menu"]').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
