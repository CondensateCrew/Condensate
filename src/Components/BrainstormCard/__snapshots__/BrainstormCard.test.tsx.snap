// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrainstormCard component should match the snapshot 1`] = `
<div
  className="bs-card"
>
  <header>
    <h4>
      What?
    </h4>
    <img
      alt="menu"
      onClick={[Function]}
      src="menu-dot.svg"
    />
  </header>
  <h5 />
  <p />
  <footer>
    <p
      className="genius-label"
    >
      Genius
    </p>
    <p
      key="ctg-tech"
    >
      Tech
    </p>
  </footer>
</div>
`;

exports[`BrainstormCard component should match the snapshot if brainstorm is not genius 1`] = `
<div
  className="bs-card"
>
  <header>
    <h4>
      What?
    </h4>
    <img
      alt="menu"
      onClick={[Function]}
      src="menu-dot.svg"
    />
  </header>
  <h5 />
  <p />
  <footer>
    <p
      key="ctg-tech"
    >
      Tech
    </p>
  </footer>
</div>
`;

exports[`BrainstormCard component should match the snapshot if brainstorm is not genius and menu is clicked 1`] = `
<div
  className="bs-card"
>
  <header>
    <h4>
      What?
    </h4>
    <MenuBlock
      id={1}
      setIsClicked={[Function]}
    />
  </header>
  <h5 />
  <p />
  <footer>
    <p
      key="ctg-tech"
    >
      Tech
    </p>
  </footer>
</div>
`;

exports[`BrainstormCard component should match the snapshot if menu is clicked 1`] = `
<div
  className="bs-card"
>
  <header>
    <h4>
      What?
    </h4>
    <MenuBlock
      id={1}
      setIsClicked={[Function]}
    />
  </header>
  <h5 />
  <p />
  <footer>
    <p
      className="genius-label"
    >
      Genius
    </p>
    <p
      key="ctg-tech"
    >
      Tech
    </p>
  </footer>
</div>
`;
